app/controllers/posts_controller.rb

```diff
@@ -11,10 +11,6 @@ class PostsController < ApplicationController
   #other then the show, new, and create action
   before_action :authorize_user, except: [:show, :new, :create]
 
-  #before_action :authorize_moderator, except: [:show]
-
-
-
```

Comment:
Let us use a single method to handle all the authorization stuff. It will make things easier to reason about, and the code less error-prone.